backend/user-service/src/controllers/auth.controller.ts

```diff
@@ -67,3 +75,91 @@ export async function hashPassword(password: string): Promise<string> {
     const saltRounds = 10
     return hash(password, saltRounds)
 }
+
+export async function getHTMLTemplate(htmlFilePath: string): Promise<string> {
+    const filePath = path.join(__dirname, htmlFilePath)
+    return await fs.readFile(filePath, 'utf8')
+}
+
+export async function sendMail(to: string, subject: string, text: string, html: string): Promise<void> {
+    const transporter = nodemailer.createTransport({
+        service: 'gmail',
+        host: 'smtp.gmail.com',
+        port: 587,
+        secure: false,
+        auth: {
+            user: process.env.NODEMAILER_EMAIL,
+            pass: process.env.NODEMAILER_PASSWORD,
+        },
+    })
+    await transporter.sendMail({
+        from: process.env.NODEMAILER_EMAIL,
+        to,
+        subject,
+        text,
+        html,
+    })
+}
+
+export function generateOTP(): string {
+    return Math.floor(100000 + Math.random() * 900000).toString()
+}
+
+export async function handleReset(request: TypedRequest<EmailVerificationDto>, response: Response): Promise<void> {
+    const createDto = EmailVerificationDto.fromRequest(request)
+    createDto.verificationToken = '0'
```

Comment:
What's the purpose of this?

Reply:
It's because the handle reset endpoint doesn't send a verification token with the body, but the `EmailVerificationDto` requires that a verification token be present, thus i'm setting it to '0' and then verifying it. Otherwise, I had to create a new type just to handle this scenario which I felt is repetitive.